File: src/sdks/manage/test/transpiled-suite/localization.test.js
/*
 * Copyright 2021 Comcast Cable Communications Management, LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 * SPDX-License-Identifier: Apache-2.0
 */
import { test, expect } from "@jest/globals";
import { Localization } from "../../build/javascript/src/firebolt-manage";
test("Localization.locality()", () => {
    return Localization.locality().then((res) => {
        expect(res).toBe("Philadelphia");
    });
});
test("Localization.countryCode()", () => {
    return Localization.countryCode().then((res) => {
        expect(res).toBe("US");
    });
});
test("Localization.language()", () => {
    return Localization.language().then((res) => {
        expect(res).toBe("en");
    });
});
test("Localization.locale()", () => {
    return Localization.locale().then((res) => {
        expect(res).toBe("en-US");
    });
});
test("Localization.locale()", () => {
    return Localization.locale().then((res) => {
        expect(res).toBe("en-US");
    });
});
test("Localization.listen()", () => {
    return Localization.listen("localityChanged", () => { }).then((res) => {
        expect(res > 0).toBe(true);
    });
});
test("Localization.once()", () => {
    return Localization.once("localityChanged", () => { }).then((res) => {
        expect(res > 0).toBe(true);
    });
});
test("Localization.clear()", () => {
    const result = Localization.clear(1);
    expect(result).toBe(undefined);
});
test("Device.postalCode()", () => {
    return Localization.postalCode().then((res) => {
        expect(res).toEqual("19103");
    });
});

File: src/cpp/test/unit/localizationTest.cpp
/*
 * If not stated otherwise in this file or this component's LICENSE file the
 * following copyright and licenses apply:
 *
 * Copyright 2025 Sky UK
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

#include "json_engine.h"
#include "localization_impl.h"
#include "mock_helper.h"

class LocalizationTest : public ::testing::Test, protected MockBase
{
protected:
    Firebolt::Localization::LocalizationImpl localizationImpl_{mockHelper};
};

TEST_F(LocalizationTest, CountryCode)
{
    mock("Localization.countryCode");

    auto result = localizationImpl_.countryCode();
    ASSERT_TRUE(result) << "error on get";

    auto expectedValue = jsonEngine.get_value("Localization.countryCode").get<std::string>();
    EXPECT_EQ(*result, expectedValue);
}

TEST_F(LocalizationTest, PreferredAudioLanguages)
{
    mock("Localization.preferredAudioLanguages");

    auto result = localizationImpl_.preferredAudioLanguages();
    ASSERT_TRUE(result) << "error on get";

    auto expectedValue = jsonEngine.get_value("Localization.preferredAudioLanguages");
    std::set<std::string> expectedSet(expectedValue.begin(), expectedValue.end());
    std::set<std::string> resultSet(result->begin(), result->end());
    EXPECT_EQ(resultSet, expectedSet);
}

TEST_F(LocalizationTest, subscribeOnCountryCodeChanged)
{
    mockSubscribe("Localization.onCountryCodeChanged");

    auto id = localizationImpl_.subscribeOnCountryCodeChanged([](auto) { std::cout << "callback\n"; });
    ASSERT_TRUE(id) << "error on subscribe ";
    EXPECT_TRUE(id.has_value()) << "error on id";
    auto result = localizationImpl_.unsubscribe(id.value_or(0));
    ASSERT_TRUE(result) << "error on unsubscribe ";
}

TEST_F(LocalizationTest, subscribeOnPreferredAudioLanguagesChanged)
{
    mockSubscribe("Localization.onPreferredAudioLanguagesChanged");

    auto id = localizationImpl_.subscribeOnPreferredAudioLanguagesChanged([](auto) { std::cout << "callback\n"; });
    ASSERT_TRUE(id) << "error on subscribe ";
    EXPECT_TRUE(id.has_value()) << "error on id";
    auto result = localizationImpl_.unsubscribe(id.value_or(0));
    ASSERT_TRUE(result) << "error on unsubscribe ";
}
